modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/OrderActions.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {CommerceServiceProvider} from 'commerce-frontend-js';
+import {openToast} from 'frontend-js-web';
+import React, {useCallback, useEffect, useState} from 'react';
+
+function OrderActions({isOpen, orderId}) {
+	const [open, setOpen] = useState(isOpen);
+	const [actions, setActions] = useState([]);
+
+	const getActions = useCallback(() => {
+		const getTransitions = open
+			? CommerceServiceProvider.DeliveryCartAPI('v1')
+					.getCartTransitionsById
+			: CommerceServiceProvider.DeliveryOrderAPI('v1')
+					.getOrderTransitionsById;
+
+		getTransitions(orderId)
+			.then((response) => {
+				setActions(response.items);
+			})
+			.catch((error) => {
+				openToast({
+					message:
+						error.message ||
+						Liferay.Language.get('an-unexpected-error-occurred'),
+					type: 'danger',
+				});
+			});
+	}, [orderId, open]);
+
+	useEffect(() => getActions(), [getActions]);
+
+	const onClick = (event, action) => {
+		event.preventDefault();
+
+		if (action.name === 'checkout') {
+			Liferay.fire('current-order-action-triggered', {id: action.name});
+		}
+		else {
+			const executeTransitions = open
+				? CommerceServiceProvider.DeliveryCartAPI('v1')
+						.executeCartTransitionsById
+				: CommerceServiceProvider.DeliveryOrderAPI('v1')
+						.executeOrderTransitionsById;
+
+			executeTransitions(orderId, action)
+				.then((response) => {
+					if (open !== response?.open) {
+						setOpen(response.open);
+					}
+					else {
+						getActions();
+					}
+
+					if (action.name === 'reorder') {
+						Liferay.fire('current-order-action-triggered', {
+							id: response.name,
+							orderId: response.orderId,
+						});
+					}
+				})
```

Comment:
why don't you make the redirect here?

Reply:
Because the Redirect fragment will contain the logic for both Order Actions and all dataset fragments.